Support `tasty-hedgehog-1.2.*`

Currently, `bv-sized`'s test suite does not support `tasty-hedgehog-1.2.0.0`, the latest Hackage release:

https://github.com/GaloisInc/bv-sized/blob/9232347a9d916c5dcbd038aed872abcb697803f8/bv-sized.cabal#L52

This poses an issue when trying to update `macaw`'s `bv-sized` submodule, as newer versions of `what4` and related libraries require `tasty-hedgehog-1.2.*` (see GaloisInc/what4#198, for instance). While it would be simple to raise `bv-sized`'s upper version bounds on `tasty-hedgehog`, doing so reveals a slew of deprecation warnings due to qfpl/tasty-hedgehog#42:

<details>

```
[1 of 1] Compiling Main             ( test/Main.hs, interpreted )

test/Main.hs:221:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
221 |   [ testProperty "and" $ bin anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:224:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
224 |   , testProperty "or" $ bin anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:227:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
227 |   , testProperty "xor" $ bin anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:230:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
230 |   , testProperty "complement" $ un anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:237:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
237 |   [ testProperty "add" $ bin anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:240:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
240 |   , testProperty "sub" $ bin anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:243:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
243 |   , testProperty "mul" $ bin anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:246:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
246 |   , testProperty "uquot" $ bin anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:249:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
249 |   , testProperty "urem" $ bin anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:252:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
252 |   , testProperty "squot-pos-denom" $ bin anyWidthGT1 BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:255:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
255 |   , testProperty "squot-neg-denom" $ bin anyWidthGT1 BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:258:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
258 |   , testProperty "srem-pos-denom" $ bin anyWidthGT1 BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:261:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
261 |   , testProperty "srem-neg-denom" $ bin anyWidthGT1 BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:264:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
264 |   , testProperty "sdiv-pos-denom" $ bin anyWidthGT1 BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:267:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
267 |   , testProperty "sdiv-neg-denom" $ bin anyWidthGT1 BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:270:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
270 |   , testProperty "smod-pos-denom" $ bin anyWidthGT1 BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:273:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
273 |   , testProperty "smod-neg-denom" $ bin anyWidthGT1 BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:276:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
276 |   , testProperty "abs" $ un anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:279:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
279 |   , testProperty "negate" $ un anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:282:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
282 |   , testProperty "signBit" $ un anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:285:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
285 |   , testProperty "signum" $ un anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:288:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
288 |   , testProperty "slt" $ binPred anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:291:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
291 |   , testProperty "sle" $ binPred anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:294:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
294 |   , testProperty "ult" $ binPred anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:297:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
297 |   , testProperty "ule" $ binPred anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:300:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
300 |   , testProperty "umin" $ bin anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:303:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
303 |   , testProperty "umax" $ bin anyWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:306:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
306 |   , testProperty "smin" $ bin anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:309:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
309 |   , testProperty "smax" $ bin anyPosWidth BV.mkBV
    |     ^^^^^^^^^^^^

test/Main.hs:350:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
350 |   [ testProperty "bitsBE" $
    |     ^^^^^^^^^^^^

test/Main.hs:352:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
352 |   , testProperty "bitsLE" $
    |     ^^^^^^^^^^^^

test/Main.hs:354:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
354 |   , testProperty "bytesBE" $
    |     ^^^^^^^^^^^^

test/Main.hs:356:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
356 |   , testProperty "bytesLE" $
    |     ^^^^^^^^^^^^

test/Main.hs:358:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
358 |   , testProperty "bytestringBE" $
    |     ^^^^^^^^^^^^

test/Main.hs:360:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
360 |   , testProperty "bytestringLE" $
    |     ^^^^^^^^^^^^

test/Main.hs:366:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
366 |   [ testProperty "asBitsBE" $
    |     ^^^^^^^^^^^^

test/Main.hs:368:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
368 |   , testProperty "asBitsLE" $
    |     ^^^^^^^^^^^^

test/Main.hs:370:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
370 |   , testProperty "asBytesBE" $
    |     ^^^^^^^^^^^^

test/Main.hs:372:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
372 |   , testProperty "asBytesLE" $
    |     ^^^^^^^^^^^^

test/Main.hs:374:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
374 |   , testProperty "asBytesBE" $
    |     ^^^^^^^^^^^^

test/Main.hs:376:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
376 |   , testProperty "asBytesLE" $
    |     ^^^^^^^^^^^^

test/Main.hs:502:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
502 |   [ testProperty "mkBV" $ wfCtor anyWidth (fmap Just . BV.mkBV)
    |     ^^^^^^^^^^^^

test/Main.hs:503:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
503 |   , testProperty "mkBVUnsigned" $ wfCtor anyWidth BV.mkBVUnsigned
    |     ^^^^^^^^^^^^

test/Main.hs:504:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
504 |   , testProperty "mkBVSigned" $ wfCtor anyPosWidth (forcePos BV.mkBVSigned)
    |     ^^^^^^^^^^^^

test/Main.hs:505:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
505 |   , testProperty "signedClamp" $ wfCtor anyPosWidth (fmap Just . forcePos BV.signedClamp)
    |     ^^^^^^^^^^^^

test/Main.hs:506:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
506 |   , testProperty "minUnsigned" $ wfCtor anyWidth (\w _ -> Just (BV.minUnsigned w))
    |     ^^^^^^^^^^^^

test/Main.hs:507:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
507 |   , testProperty "maxUnsigned" $ wfCtor anyWidth (\w _ -> Just (BV.maxUnsigned w))
    |     ^^^^^^^^^^^^

test/Main.hs:508:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
508 |   , testProperty "minSigned" $ wfCtor anyPosWidth (\w _ -> Just (forcePos BV.minSigned w))
    |     ^^^^^^^^^^^^

test/Main.hs:509:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
509 |   , testProperty "maxSigned" $ wfCtor anyPosWidth (\w _ -> Just (forcePos BV.maxSigned w))
    |     ^^^^^^^^^^^^

test/Main.hs:510:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
510 |   , testProperty "bool" $ wfCtor' knownNat (BV.bool . odd)
    |     ^^^^^^^^^^^^

test/Main.hs:511:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
511 |   , testProperty "word8" $ wfCtor' knownNat (BV.word8 . fromInteger)
    |     ^^^^^^^^^^^^

test/Main.hs:512:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
512 |   , testProperty "word16" $ wfCtor' knownNat (BV.word16 . fromInteger)
    |     ^^^^^^^^^^^^

test/Main.hs:513:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
513 |   , testProperty "word32" $ wfCtor' knownNat (BV.word32 . fromInteger)
    |     ^^^^^^^^^^^^

test/Main.hs:514:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
514 |   , testProperty "word64" $ wfCtor' knownNat (BV.word64 . fromInteger)
    |     ^^^^^^^^^^^^

test/Main.hs:515:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
515 |   , testProperty "int8" $ wfCtor' knownNat (BV.int8 . fromInteger)
    |     ^^^^^^^^^^^^

test/Main.hs:516:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
516 |   , testProperty "int16" $ wfCtor' knownNat (BV.int16 . fromInteger)
    |     ^^^^^^^^^^^^

test/Main.hs:517:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
517 |   , testProperty "int32" $ wfCtor' knownNat (BV.int32 . fromInteger)
    |     ^^^^^^^^^^^^

test/Main.hs:518:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
518 |   , testProperty "int64" $ wfCtor' knownNat (BV.int64 . fromInteger)
    |     ^^^^^^^^^^^^

test/Main.hs:519:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
519 |   , testProperty "and" $ wfBinary anyWidth (const BV.and)
    |     ^^^^^^^^^^^^

test/Main.hs:520:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
520 |   , testProperty "or" $ wfBinary anyWidth (const BV.or)
    |     ^^^^^^^^^^^^

test/Main.hs:521:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
521 |   , testProperty "xor" $ wfBinary anyWidth (const BV.xor)
    |     ^^^^^^^^^^^^

test/Main.hs:522:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
522 |   , testProperty "complement" $ wfUnary anyWidth BV.complement
    |     ^^^^^^^^^^^^

test/Main.hs:523:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
523 |   , testProperty "shl" $ wfBinaryN anyWidth BV.shl
    |     ^^^^^^^^^^^^

test/Main.hs:524:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
524 |   , testProperty "ashr" $ wfBinaryN anyPosWidth (forcePos BV.ashr)
    |     ^^^^^^^^^^^^

test/Main.hs:525:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
525 |   , testProperty "lshr" $ wfBinaryN anyWidth BV.lshr
    |     ^^^^^^^^^^^^

test/Main.hs:526:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
526 |   , testProperty "rotateL" $ wfBinaryN anyWidth BV.rotateL
    |     ^^^^^^^^^^^^

test/Main.hs:527:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
527 |   , testProperty "rotateR" $ wfBinaryN anyWidth BV.rotateR
    |     ^^^^^^^^^^^^

test/Main.hs:528:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
528 |   , testProperty "bit" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:534:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
534 |   , testProperty "bit'" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:540:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
540 |   , testProperty "setBit" $ wfBit anyPosWidth (const BV.setBit)
    |     ^^^^^^^^^^^^

test/Main.hs:541:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
541 |   , testProperty "setBit'" $ wfBitN anyPosWidth BV.setBit'
    |     ^^^^^^^^^^^^

test/Main.hs:542:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
542 |   , testProperty "clearBit" $ wfBit anyPosWidth BV.clearBit
    |     ^^^^^^^^^^^^

test/Main.hs:543:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
543 |   , testProperty "clearBit'" $ wfBitN anyPosWidth BV.clearBit'
    |     ^^^^^^^^^^^^

test/Main.hs:544:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
544 |   , testProperty "complementBit" $ wfBit anyPosWidth (const BV.complementBit)
    |     ^^^^^^^^^^^^

test/Main.hs:545:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
545 |   , testProperty "complementBit'" $ wfBitN anyPosWidth BV.complementBit'
    |     ^^^^^^^^^^^^

test/Main.hs:546:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
546 |   , testProperty "popCount" $ wfUnary anyWidth (const BV.popCount)
    |     ^^^^^^^^^^^^

test/Main.hs:547:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
547 |   , testProperty "ctz" $ wfUnary anyWidth BV.ctz
    |     ^^^^^^^^^^^^

test/Main.hs:548:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
548 |   , testProperty "clz" $ wfUnary anyWidth BV.clz
    |     ^^^^^^^^^^^^

test/Main.hs:549:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
549 |   , testProperty "truncBits" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:556:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
556 |   , testProperty "add" $ wfBinary anyWidth BV.add
    |     ^^^^^^^^^^^^

test/Main.hs:557:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
557 |   , testProperty "sub" $ wfBinary anyWidth BV.sub
    |     ^^^^^^^^^^^^

test/Main.hs:558:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
558 |   , testProperty "mul" $ wfBinary anyWidth BV.mul
    |     ^^^^^^^^^^^^

test/Main.hs:559:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
559 |   , testProperty "uquot" $ wfBinaryDiv anyPosWidth (const BV.uquot)
    |     ^^^^^^^^^^^^

test/Main.hs:560:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
560 |   , testProperty "urem" $ wfBinaryDiv anyPosWidth (const BV.urem)
    |     ^^^^^^^^^^^^

test/Main.hs:561:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
561 |   , testProperty "squot" $ wfBinaryDiv anyPosWidth (forcePos BV.squot)
    |     ^^^^^^^^^^^^

test/Main.hs:562:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
562 |   , testProperty "srem" $ wfBinaryDiv anyPosWidth (forcePos BV.srem)
    |     ^^^^^^^^^^^^

test/Main.hs:563:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
563 |   , testProperty "sdiv" $ wfBinaryDiv anyPosWidth (forcePos BV.sdiv)
    |     ^^^^^^^^^^^^

test/Main.hs:564:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
564 |   , testProperty "smod" $ wfBinaryDiv anyPosWidth (forcePos BV.smod)
    |     ^^^^^^^^^^^^

test/Main.hs:565:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
565 |   , testProperty "abs" $ wfUnary anyPosWidth (forcePos BV.abs)
    |     ^^^^^^^^^^^^

test/Main.hs:566:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
566 |   , testProperty "negate" $ wfUnary anyWidth BV.negate
    |     ^^^^^^^^^^^^

test/Main.hs:567:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
567 |   , testProperty "signBit" $ wfUnary anyPosWidth (forcePos BV.signBit)
    |     ^^^^^^^^^^^^

test/Main.hs:568:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
568 |   , testProperty "signum" $ wfUnary anyPosWidth (forcePos BV.signum)
    |     ^^^^^^^^^^^^

test/Main.hs:569:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
569 |   , testProperty "umin" $ wfBinary anyWidth (const BV.umin)
    |     ^^^^^^^^^^^^

test/Main.hs:570:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
570 |   , testProperty "umax" $ wfBinary anyWidth (const BV.umax)
    |     ^^^^^^^^^^^^

test/Main.hs:571:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
571 |   , testProperty "smin" $ wfBinary anyPosWidth (forcePos BV.smin)
    |     ^^^^^^^^^^^^

test/Main.hs:572:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
572 |   , testProperty "smax" $ wfBinary anyPosWidth (forcePos BV.smax)
    |     ^^^^^^^^^^^^

test/Main.hs:573:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
573 |   , testProperty "concat" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:581:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
581 |   , testProperty "select" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:591:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
591 |   , testProperty "select'" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:599:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
599 |   , testProperty "zext" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:606:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
606 |   , testProperty "sext" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:613:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
613 |   , testProperty "trunc" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:620:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
620 |   , testProperty "trunc'" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:625:21: warning: [-Wdeprecations]
    In the use of ‘trunc'’
    (imported from Data.BitVector.Sized, but defined in bv-sized-1.0.4.99:Data.BitVector.Sized.Internal):
    Deprecated: "Use zresize instead"
    |
625 |       let BV.BV x = BV.trunc' w' bv
    |                     ^^^^^^^^^

test/Main.hs:627:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
627 |   , testProperty "zresize" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:634:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
634 |   , testProperty "sresize" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:642:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
642 |   , testProperty "mulWide" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:650:7: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
650 |     , testProperty "succUnsigned" $ wfUnaryMaybe anyWidth BV.succUnsigned
    |       ^^^^^^^^^^^^

test/Main.hs:651:7: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
651 |     , testProperty "succSigned" $ wfUnaryMaybe anyPosWidth (forcePos BV.succUnsigned)
    |       ^^^^^^^^^^^^

test/Main.hs:652:7: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
652 |     , testProperty "predUnsigned" $ wfUnaryMaybe anyWidth BV.predUnsigned
    |       ^^^^^^^^^^^^

test/Main.hs:653:7: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
653 |     , testProperty "predSigned" $ wfUnaryMaybe anyPosWidth (forcePos BV.predUnsigned)
    |       ^^^^^^^^^^^^

test/Main.hs:679:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
679 |   [ testProperty "random unsigned well-formed" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:682:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
682 |   , testProperty "random signed well-formed" $ property $ do
    |     ^^^^^^^^^^^^

test/Main.hs:685:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
685 |   , testProperty "randomR unsigned well-formed and in bounds" $
    |     ^^^^^^^^^^^^

test/Main.hs:687:5: warning: [-Wdeprecations]
    In the use of ‘testProperty’ (imported from Test.Tasty.Hedgehog):
    Deprecated: "testProperty will cause Hedgehog to provide incorrect instructions for re-checking properties"
    |
687 |   , testProperty "randomR signed well-formed and in bounds" $
    |     ^^^^^^^^^^^^
```
</details>

There are two possible ways to fix these deprecation warnings:

1. Define our own version of `testProperty` in terms of [`testPropertyNamed`](https://hackage.haskell.org/package/tasty-hedgehog-1.2.0.0/docs/Test-Tasty-Hedgehog.html#v:testPropertyNamed). This is the approach that `what4` takes in GaloisInc/what4#198.
2. Change all properties to top-level values and change every use of `testProperty` to `testPropertyNamed`. This is the approach that `parameterized-utils` takes in GaloisInc/parameterized-utils#131.

@benjaminselfridge, do you have a preference on which solution I should implement? I'd be OK with either one, although (2) would take more work to achieve.

--- a/bv-sized.cabal
+++ b/bv-sized.cabal
@@ -49,6 +49,6 @@ test-suite bv-sized-tests
                        MonadRandom >= 0.5.3 && < 0.6,
                        parameterized-utils,
                        tasty >= 1.2.3 && < 1.5,
-                       tasty-hedgehog >= 1.0.0.2 && < 1.2
+                       tasty-hedgehog >= 1.2 && < 1.3
   default-language:    Haskell2010
   ghc-options:         -Wall -Wcompat

--- a/test/Main.hs
+++ b/test/Main.hs
@@ -29,8 +29,25 @@ import Data.Maybe (isJust, fromJust)
 import Data.Parameterized.NatRepr
 import Data.Parameterized.Some
 import Data.Parameterized.Pair
+import Data.String ( fromString )
 import Data.Word
 import Numeric.Natural
+
+-- | This is a wrapper around 'testPropertyNamed' that somewhat addresses a
+-- deprecation warning.  Newer versions of the test library deprecated
+-- 'testProperty' in favor of 'testPropertyNamed', which is intended to provide
+-- better human-readable instructions to reproduce test failures.
+--
+-- However, doing that requires all properties to have a top-level definition
+-- corresponding to them, which we do not here. We just re-use the string
+-- description as the test name, which will produce incorrect instructions to
+-- reproduce failures, but will provide sufficient context clues for a developer
+-- to figure out which test failed.
+--
+-- The alternative is to refactor all of the tests into top-level properties to
+-- use the API as intended.
+testPropertyString :: String -> Property -> TestTree
+testPropertyString desc = testPropertyNamed desc (fromString desc)
 
 ----------------------------------------
 -- Utilities
@@ -218,95 +235,95 @@ genPair gen gen' = do
 
 bitwiseHomTests :: TestTree
 bitwiseHomTests = testGroup "bitwise homomorphisms tests"
-  [ testProperty "and" $ bin anyWidth BV.mkBV
+  [ testPropertyString "and" $ bin anyWidth BV.mkBV
     largeUnsigned largeUnsigned
     (const (Bits..&.)) (const BV.and)
-  , testProperty "or" $ bin anyWidth BV.mkBV
+  , testPropertyString "or" $ bin anyWidth BV.mkBV
     largeUnsigned largeUnsigned
     (const (Bits..|.)) (const BV.or)
-  , testProperty "xor" $ bin anyWidth BV.mkBV
+  , testPropertyString "xor" $ bin anyWidth BV.mkBV
     largeUnsigned largeUnsigned
     (const Bits.xor) (const BV.xor)
-  , testProperty "complement" $ un anyWidth BV.mkBV
+  , testPropertyString "complement" $ un anyWidth BV.mkBV
     largeUnsigned
     (const Bits.complement) BV.complement
   ]
 
 arithHomTests :: TestTree
 arithHomTests = testGroup "arithmetic homomorphisms tests"
-  [ testProperty "add" $ bin anyWidth BV.mkBV
+  [ testPropertyString "add" $ bin anyWidth BV.mkBV
     largeSigned largeSigned
     (const (+)) BV.add
-  , testProperty "sub" $ bin anyWidth BV.mkBV
+  , testPropertyString "sub" $ bin anyWidth BV.mkBV
     largeSigned largeSigned
     (const (-)) BV.sub
-  , testProperty "mul" $ bin anyWidth BV.mkBV
+  , testPropertyString "mul" $ bin anyWidth BV.mkBV
     largeSigned largeSigned
     (const (*)) BV.mul
-  , testProperty "uquot" $ bin anyPosWidth BV.mkBV
+  , testPropertyString "uquot" $ bin anyPosWidth BV.mkBV
     unsigned unsignedPos
     (const quot) (const BV.uquot)
-  , testProperty "urem" $ bin anyPosWidth BV.mkBV
+  , testPropertyString "urem" $ bin anyPosWidth BV.mkBV
     unsigned unsignedPos
     (const rem) (const BV.urem)
-  , testProperty "squot-pos-denom" $ bin anyWidthGT1 BV.mkBV
+  , testPropertyString "squot-pos-denom" $ bin anyWidthGT1 BV.mkBV
     signed signedPos
     (const quot) (forcePos BV.squot)
-  , testProperty "squot-neg-denom" $ bin anyWidthGT1 BV.mkBV
+  , testPropertyString "squot-neg-denom" $ bin anyWidthGT1 BV.mkBV
     signed signedNeg
     (const quot) (forcePos BV.squot)
-  , testProperty "srem-pos-denom" $ bin anyWidthGT1 BV.mkBV
+  , testPropertyString "srem-pos-denom" $ bin anyWidthGT1 BV.mkBV
     signed signedPos
     (const rem) (forcePos BV.srem)
-  , testProperty "srem-neg-denom" $ bin anyWidthGT1 BV.mkBV
+  , testPropertyString "srem-neg-denom" $ bin anyWidthGT1 BV.mkBV
     signed signedNeg
     (const rem) (forcePos BV.srem)
-  , testProperty "sdiv-pos-denom" $ bin anyWidthGT1 BV.mkBV
+  , testPropertyString "sdiv-pos-denom" $ bin anyWidthGT1 BV.mkBV
     signed signedPos
     (const div) (forcePos BV.sdiv)
-  , testProperty "sdiv-neg-denom" $ bin anyWidthGT1 BV.mkBV
+  , testPropertyString "sdiv-neg-denom" $ bin anyWidthGT1 BV.mkBV
     signed signedNeg
     (const div) (forcePos BV.sdiv)
-  , testProperty "smod-pos-denom" $ bin anyWidthGT1 BV.mkBV
+  , testPropertyString "smod-pos-denom" $ bin anyWidthGT1 BV.mkBV
     signed signedPos
     (const mod) (forcePos BV.smod)
-  , testProperty "smod-neg-denom" $ bin anyWidthGT1 BV.mkBV
+  , testPropertyString "smod-neg-denom" $ bin anyWidthGT1 BV.mkBV
     signed signedNeg
     (const mod) (forcePos BV.smod)
-  , testProperty "abs" $ un anyPosWidth BV.mkBV
+  , testPropertyString "abs" $ un anyPosWidth BV.mkBV
     signed
     (const abs) (forcePos BV.abs)
-  , testProperty "negate" $ un anyPosWidth BV.mkBV
+  , testPropertyString "negate" $ un anyPosWidth BV.mkBV
     largeSigned
     (const negate) BV.negate
-  , testProperty "signBit" $ un anyPosWidth BV.mkBV
+  , testPropertyString "signBit" $ un anyPosWidth BV.mkBV
     signed
     (\_ a -> if a < 0 then 1 else 0) (forcePos BV.signBit)
-  , testProperty "signum" $ un anyPosWidth BV.mkBV
+  , testPropertyString "signum" $ un anyPosWidth BV.mkBV
     signed
     (\_ a -> signum a) (forcePos BV.signum)
-  , testProperty "slt" $ binPred anyPosWidth BV.mkBV
+  , testPropertyString "slt" $ binPred anyPosWidth BV.mkBV
     signed signed
     (const (<)) (forcePos BV.slt)
-  , testProperty "sle" $ binPred anyPosWidth BV.mkBV
+  , testPropertyString "sle" $ binPred anyPosWidth BV.mkBV
     signed signed
     (const (<=)) (forcePos BV.sle)
-  , testProperty "ult" $ binPred anyWidth BV.mkBV
+  , testPropertyString "ult" $ binPred anyWidth BV.mkBV
     unsigned unsigned
     (const (<)) (const BV.ult)
-  , testProperty "ule" $ binPred anyWidth BV.mkBV
+  , testPropertyString "ule" $ binPred anyWidth BV.mkBV
     unsigned unsigned
     (const (<=)) (const BV.ule)
-  , testProperty "umin" $ bin anyWidth BV.mkBV
+  , testPropertyString "umin" $ bin anyWidth BV.mkBV
     unsigned unsigned
     (const min) (const BV.umin)
-  , testProperty "umax" $ bin anyWidth BV.mkBV
+  , testPropertyString "umax" $ bin anyWidth BV.mkBV
     unsigned unsigned
     (const max) (const BV.umax)
-  , testProperty "smin" $ bin anyPosWidth BV.mkBV
+  , testPropertyString "smin" $ bin anyPosWidth BV.mkBV
     signed signed
     (const min) (forcePos BV.smin)
-  , testProperty "smax" $ bin anyPosWidth BV.mkBV
+  , testPropertyString "smax" $ bin anyPosWidth BV.mkBV
     signed signed
     (const max) (forcePos BV.smax)
   ]
@@ -347,33 +364,33 @@ deserTest genA lenA de ser = property $ do
 
 serdeTests :: TestTree
 serdeTests = testGroup "serialization/deseriallization tests"
-  [ testProperty "bitsBE" $
+  [ testPropertyString "bitsBE" $
     serdeTest anyWidth (\w bv -> Just (BV.asBitsBE w bv)) BV.bitsBE
-  , testProperty "bitsLE" $
+  , testPropertyString "bitsLE" $
     serdeTest anyWidth (\w bv -> Just (BV.asBitsLE w bv)) BV.bitsLE
-  , testProperty "bytesBE" $
+  , testPropertyString "bytesBE" $
     serdeTest byteWidth BV.asBytesBE BV.bytesBE
-  , testProperty "bytesLE" $
+  , testPropertyString "bytesLE" $
     serdeTest byteWidth BV.asBytesLE BV.bytesLE
-  , testProperty "bytestringBE" $
+  , testPropertyString "bytestringBE" $
     serdeTest byteWidth BV.asBytestringBE BV.bytestringBE
-  , testProperty "bytestringLE" $
+  , testPropertyString "bytestringLE" $
     serdeTest byteWidth BV.asBytestringLE BV.bytestringLE
   ]
 
 deserTests :: TestTree
 deserTests = testGroup "deserialization/serialization tests"
-  [ testProperty "asBitsBE" $
+  [ testPropertyString "asBitsBE" $
     deserTest bits length BV.bitsBE (\w bv -> Just (BV.asBitsBE w bv))
-  , testProperty "asBitsLE" $
+  , testPropertyString "asBitsLE" $
     deserTest bits length BV.bitsLE (\w bv -> Just (BV.asBitsLE w bv))
-  , testProperty "asBytesBE" $
+  , testPropertyString "asBytesBE" $
     deserTest bytes ((*8) . length) BV.bytesBE BV.asBytesBE
-  , testProperty "asBytesLE" $
+  , testPropertyString "asBytesLE" $
     deserTest bytes ((*8) . length) BV.bytesLE BV.asBytesLE
-  , testProperty "asBytesBE" $
+  , testPropertyString "asBytesBE" $
     deserTest (BS.pack <$> bytes) ((*8) . BS.length) BV.bytestringBE BV.asBytestringBE
-  , testProperty "asBytesLE" $
+  , testPropertyString "asBytesLE" $
     deserTest (BS.pack <$> bytes) ((*8) . BS.length) BV.bytestringLE BV.asBytestringLE
   ]
 
@@ -499,78 +516,78 @@ wfBitN genW f = property $ do
 
 wellFormedTests :: TestTree
 wellFormedTests = testGroup "well-formedness tests"
-  [ testProperty "mkBV" $ wfCtor anyWidth (fmap Just . BV.mkBV)
-  , testProperty "mkBVUnsigned" $ wfCtor anyWidth BV.mkBVUnsigned
-  , testProperty "mkBVSigned" $ wfCtor anyPosWidth (forcePos BV.mkBVSigned)
-  , testProperty "signedClamp" $ wfCtor anyPosWidth (fmap Just . forcePos BV.signedClamp)
-  , testProperty "minUnsigned" $ wfCtor anyWidth (\w _ -> Just (BV.minUnsigned w))
-  , testProperty "maxUnsigned" $ wfCtor anyWidth (\w _ -> Just (BV.maxUnsigned w))
-  , testProperty "minSigned" $ wfCtor anyPosWidth (\w _ -> Just (forcePos BV.minSigned w))
-  , testProperty "maxSigned" $ wfCtor anyPosWidth (\w _ -> Just (forcePos BV.maxSigned w))
-  , testProperty "bool" $ wfCtor' knownNat (BV.bool . odd)
-  , testProperty "word8" $ wfCtor' knownNat (BV.word8 . fromInteger)
-  , testProperty "word16" $ wfCtor' knownNat (BV.word16 . fromInteger)
-  , testProperty "word32" $ wfCtor' knownNat (BV.word32 . fromInteger)
-  , testProperty "word64" $ wfCtor' knownNat (BV.word64 . fromInteger)
-  , testProperty "int8" $ wfCtor' knownNat (BV.int8 . fromInteger)
-  , testProperty "int16" $ wfCtor' knownNat (BV.int16 . fromInteger)
-  , testProperty "int32" $ wfCtor' knownNat (BV.int32 . fromInteger)
-  , testProperty "int64" $ wfCtor' knownNat (BV.int64 . fromInteger)
-  , testProperty "and" $ wfBinary anyWidth (const BV.and)
-  , testProperty "or" $ wfBinary anyWidth (const BV.or)
-  , testProperty "xor" $ wfBinary anyWidth (const BV.xor)
-  , testProperty "complement" $ wfUnary anyWidth BV.complement
-  , testProperty "shl" $ wfBinaryN anyWidth BV.shl
-  , testProperty "ashr" $ wfBinaryN anyPosWidth (forcePos BV.ashr)
-  , testProperty "lshr" $ wfBinaryN anyWidth BV.lshr
-  , testProperty "rotateL" $ wfBinaryN anyWidth BV.rotateL
-  , testProperty "rotateR" $ wfBinaryN anyWidth BV.rotateR
-  , testProperty "bit" $ property $ do
+  [ testPropertyString "mkBV" $ wfCtor anyWidth (fmap Just . BV.mkBV)
+  , testPropertyString "mkBVUnsigned" $ wfCtor anyWidth BV.mkBVUnsigned
+  , testPropertyString "mkBVSigned" $ wfCtor anyPosWidth (forcePos BV.mkBVSigned)
+  , testPropertyString "signedClamp" $ wfCtor anyPosWidth (fmap Just . forcePos BV.signedClamp)
+  , testPropertyString "minUnsigned" $ wfCtor anyWidth (\w _ -> Just (BV.minUnsigned w))
+  , testPropertyString "maxUnsigned" $ wfCtor anyWidth (\w _ -> Just (BV.maxUnsigned w))
+  , testPropertyString "minSigned" $ wfCtor anyPosWidth (\w _ -> Just (forcePos BV.minSigned w))
+  , testPropertyString "maxSigned" $ wfCtor anyPosWidth (\w _ -> Just (forcePos BV.maxSigned w))
+  , testPropertyString "bool" $ wfCtor' knownNat (BV.bool . odd)
+  , testPropertyString "word8" $ wfCtor' knownNat (BV.word8 . fromInteger)
+  , testPropertyString "word16" $ wfCtor' knownNat (BV.word16 . fromInteger)
+  , testPropertyString "word32" $ wfCtor' knownNat (BV.word32 . fromInteger)
+  , testPropertyString "word64" $ wfCtor' knownNat (BV.word64 . fromInteger)
+  , testPropertyString "int8" $ wfCtor' knownNat (BV.int8 . fromInteger)
+  , testPropertyString "int16" $ wfCtor' knownNat (BV.int16 . fromInteger)
+  , testPropertyString "int32" $ wfCtor' knownNat (BV.int32 . fromInteger)
+  , testPropertyString "int64" $ wfCtor' knownNat (BV.int64 . fromInteger)
+  , testPropertyString "and" $ wfBinary anyWidth (const BV.and)
+  , testPropertyString "or" $ wfBinary anyWidth (const BV.or)
+  , testPropertyString "xor" $ wfBinary anyWidth (const BV.xor)
+  , testPropertyString "complement" $ wfUnary anyWidth BV.complement
+  , testPropertyString "shl" $ wfBinaryN anyWidth BV.shl
+  , testPropertyString "ashr" $ wfBinaryN anyPosWidth (forcePos BV.ashr)
+  , testPropertyString "lshr" $ wfBinaryN anyWidth BV.lshr
+  , testPropertyString "rotateL" $ wfBinaryN anyWidth BV.rotateL
+  , testPropertyString "rotateR" $ wfBinaryN anyWidth BV.rotateR
+  , testPropertyString "bit" $ property $ do
       Some w <- forAll anyPosWidth
       NatReprLt i <- forAll (natReprLt w)
 
       let BV.BV x = BV.bit w i
       checkBounds x w
-  , testProperty "bit'" $ property $ do
+  , testPropertyString "bit'" $ property $ do
       Some w <- forAll anyPosWidth
       n <- forAll $ Gen.integral $ Range.linear 0 (2 * natValue w)
 
       let BV.BV x = BV.bit' w n
       checkBounds x w
-  , testProperty "setBit" $ wfBit anyPosWidth (const BV.setBit)
-  , testProperty "setBit'" $ wfBitN anyPosWidth BV.setBit'
-  , testProperty "clearBit" $ wfBit anyPosWidth BV.clearBit
-  , testProperty "clearBit'" $ wfBitN anyPosWidth BV.clearBit'
-  , testProperty "complementBit" $ wfBit anyPosWidth (const BV.complementBit)
-  , testProperty "complementBit'" $ wfBitN anyPosWidth BV.complementBit'
-  , testProperty "popCount" $ wfUnary anyWidth (const BV.popCount)
-  , testProperty "ctz" $ wfUnary anyWidth BV.ctz
-  , testProperty "clz" $ wfUnary anyWidth BV.clz
-  , testProperty "truncBits" $ property $ do
+  , testPropertyString "setBit" $ wfBit anyPosWidth (const BV.setBit)
+  , testPropertyString "setBit'" $ wfBitN anyPosWidth BV.setBit'
+  , testPropertyString "clearBit" $ wfBit anyPosWidth BV.clearBit
+  , testPropertyString "clearBit'" $ wfBitN anyPosWidth BV.clearBit'
+  , testPropertyString "complementBit" $ wfBit anyPosWidth (const BV.complementBit)
+  , testPropertyString "complementBit'" $ wfBitN anyPosWidth BV.complementBit'
+  , testPropertyString "popCount" $ wfUnary anyWidth (const BV.popCount)
+  , testPropertyString "ctz" $ wfUnary anyWidth BV.ctz
+  , testPropertyString "clz" $ wfUnary anyWidth BV.clz
+  , testPropertyString "truncBits" $ property $ do
       Some w <- forAll anyWidth
       bv <- BV.mkBV w <$> forAll (unsigned w)
       n <- forAll $ Gen.integral $ Range.linear 0 (2 * natValue w)
 
       let BV.BV x = BV.truncBits n bv
       checkBounds x w
-  , testProperty "add" $ wfBinary anyWidth BV.add
-  , testProperty "sub" $ wfBinary anyWidth BV.sub
-  , testProperty "mul" $ wfBinary anyWidth BV.mul
-  , testProperty "uquot" $ wfBinaryDiv anyPosWidth (const BV.uquot)
-  , testProperty "urem" $ wfBinaryDiv anyPosWidth (const BV.urem)
-  , testProperty "squot" $ wfBinaryDiv anyPosWidth (forcePos BV.squot)
-  , testProperty "srem" $ wfBinaryDiv anyPosWidth (forcePos BV.srem)
-  , testProperty "sdiv" $ wfBinaryDiv anyPosWidth (forcePos BV.sdiv)
-  , testProperty "smod" $ wfBinaryDiv anyPosWidth (forcePos BV.smod)
-  , testProperty "abs" $ wfUnary anyPosWidth (forcePos BV.abs)
-  , testProperty "negate" $ wfUnary anyWidth BV.negate
-  , testProperty "signBit" $ wfUnary anyPosWidth (forcePos BV.signBit)
-  , testProperty "signum" $ wfUnary anyPosWidth (forcePos BV.signum)
-  , testProperty "umin" $ wfBinary anyWidth (const BV.umin)
-  , testProperty "umax" $ wfBinary anyWidth (const BV.umax)
-  , testProperty "smin" $ wfBinary anyPosWidth (forcePos BV.smin)
-  , testProperty "smax" $ wfBinary anyPosWidth (forcePos BV.smax)
-  , testProperty "concat" $ property $ do
+  , testPropertyString "add" $ wfBinary anyWidth BV.add
+  , testPropertyString "sub" $ wfBinary anyWidth BV.sub
+  , testPropertyString "mul" $ wfBinary anyWidth BV.mul
+  , testPropertyString "uquot" $ wfBinaryDiv anyPosWidth (const BV.uquot)
+  , testPropertyString "urem" $ wfBinaryDiv anyPosWidth (const BV.urem)
+  , testPropertyString "squot" $ wfBinaryDiv anyPosWidth (forcePos BV.squot)
+  , testPropertyString "srem" $ wfBinaryDiv anyPosWidth (forcePos BV.srem)
+  , testPropertyString "sdiv" $ wfBinaryDiv anyPosWidth (forcePos BV.sdiv)
+  , testPropertyString "smod" $ wfBinaryDiv anyPosWidth (forcePos BV.smod)
+  , testPropertyString "abs" $ wfUnary anyPosWidth (forcePos BV.abs)
+  , testPropertyString "negate" $ wfUnary anyWidth BV.negate
+  , testPropertyString "signBit" $ wfUnary anyPosWidth (forcePos BV.signBit)
+  , testPropertyString "signum" $ wfUnary anyPosWidth (forcePos BV.signum)
+  , testPropertyString "umin" $ wfBinary anyWidth (const BV.umin)
+  , testPropertyString "umax" $ wfBinary anyWidth (const BV.umax)
+  , testPropertyString "smin" $ wfBinary anyPosWidth (forcePos BV.smin)
+  , testPropertyString "smax" $ wfBinary anyPosWidth (forcePos BV.smax)
+  , testPropertyString "concat" $ property $ do
       Some w <- forAll anyWidth
       Some w' <- forAll anyWidth
       bv <- BV.mkBV w <$> forAll (unsigned w)
@@ -578,7 +595,7 @@ wellFormedTests = testGroup "well-formedness tests"
 
       let BV.BV x = BV.concat w w' bv bv'
       checkBounds x (w `addNat` w')
-  , testProperty "select" $ property $ do
+  , testPropertyString "select" $ property $ do
       Some w <- forAll anyWidth
       bv <- BV.mkBV w <$> forAll (unsigned w)
       NatReprLte ix <- forAll (natReprLte w)
@@ -588,7 +605,7 @@ wellFormedTests = testGroup "well-formedness tests"
 
       let BV.BV x = BV.select ix w' bv
       checkBounds x w'
-  , testProperty "select'" $ property $ do
+  , testPropertyString "select'" $ property $ do
       Some w <- forAll anyWidth
       Some w' <- forAll anyWidth
       bv <- BV.mkBV w <$> forAll (unsigned w)
@@ -596,42 +613,42 @@ wellFormedTests = testGroup "well-formedness tests"
 
       let BV.BV x = BV.select' n w' bv
       checkBounds x w'
-  , testProperty "zext" $ property $ do
+  , testPropertyString "zext" $ property $ do
       Some w' <- forAll anyPosWidth
       NatReprLt w <- forAll (natReprLt w')
       bv <- BV.mkBV w <$> forAll (unsigned w)
 
       let BV.BV x = BV.zext w' bv
       checkBounds x w'
-  , testProperty "sext" $ property $ do
+  , testPropertyString "sext" $ property $ do
       Some w' <- forAll anyWidthGT1
       NatReprPosLt w <- forAll (natReprPosLt w')
       bv <- BV.mkBV w <$> forAll (unsigned w)
 
       let BV.BV x = BV.sext w w' bv
       checkBounds x w'
-  , testProperty "trunc" $ property $ do
+  , testPropertyString "trunc" $ property $ do
       Some w <- forAll anyPosWidth
       NatReprLt w' <- forAll (natReprLt w)
       bv <- BV.mkBV w <$> forAll (unsigned w)
 
       let BV.BV x = BV.trunc w' bv
       checkBounds x w'
-  , testProperty "trunc'" $ property $ do
+  , testPropertyString "trunc'" $ property $ do
       Some w <- forAll anyWidth
       Some w' <- forAll anyWidth
       bv <- BV.mkBV w <$> forAll (unsigned w)
 
       let BV.BV x = BV.trunc' w' bv
       checkBounds x w'
-  , testProperty "zresize" $ property $ do
+  , testPropertyString "zresize" $ property $ do
       Some w <- forAll anyWidth
       Some w' <- forAll anyWidth
       bv <- BV.mkBV w <$> forAll (unsigned w)
 
       let BV.BV x = BV.zresize w' bv
       checkBounds x w'
-  , testProperty "sresize" $ property $ do
+  , testPropertyString "sresize" $ property $ do
       Some w <- forAll anyPosWidth
       Just LeqProof <- return $ isPosNat w
       Some w' <- forAll anyWidth
@@ -639,7 +656,7 @@ wellFormedTests = testGroup "well-formedness tests"
 
       let BV.BV x = BV.sresize w w' bv
       checkBounds x w'
-  , testProperty "mulWide" $ property $ do
+  , testPropertyString "mulWide" $ property $ do
       Some w <- forAll anyWidth
       Some w' <- forAll anyWidth
       bv <- BV.mkBV w <$> forAll (unsigned w)
@@ -647,10 +664,10 @@ wellFormedTests = testGroup "well-formedness tests"
 
       let BV.BV x = BV.mulWide w w' bv bv'
       checkBounds x (w `addNat` w')
-    , testProperty "succUnsigned" $ wfUnaryMaybe anyWidth BV.succUnsigned
-    , testProperty "succSigned" $ wfUnaryMaybe anyPosWidth (forcePos BV.succUnsigned)
-    , testProperty "predUnsigned" $ wfUnaryMaybe anyWidth BV.predUnsigned
-    , testProperty "predSigned" $ wfUnaryMaybe anyPosWidth (forcePos BV.predUnsigned)
+    , testPropertyString "succUnsigned" $ wfUnaryMaybe anyWidth BV.succUnsigned
+    , testPropertyString "succSigned" $ wfUnaryMaybe anyPosWidth (forcePos BV.succUnsigned)
+    , testPropertyString "predUnsigned" $ wfUnaryMaybe anyWidth BV.predUnsigned
+    , testPropertyString "predSigned" $ wfUnaryMaybe anyPosWidth (forcePos BV.predUnsigned)
     ]
 
 testRandomR :: (Ord (f w), Random (f w), Show (f w), Show a)
@@ -676,15 +693,15 @@ testRandomR w mk gen = property $ do
 
 randomTests :: TestTree
 randomTests = testGroup "tests for random generation"
-  [ testProperty "random unsigned well-formed" $ property $ do
+  [ testPropertyString "random unsigned well-formed" $ property $ do
       BV.UnsignedBV (BV.BV x) :: BV.UnsignedBV 32 <- liftIO $ getRandom
       checkBounds x (knownNat @32)
-  , testProperty "random signed well-formed" $ property $ do
+  , testPropertyString "random signed well-formed" $ property $ do
       BV.SignedBV (BV.BV x) :: BV.SignedBV 32 <- liftIO $ getRandom
       checkBounds x (knownNat @32)
-  , testProperty "randomR unsigned well-formed and in bounds" $
+  , testPropertyString "randomR unsigned well-formed and in bounds" $
     testRandomR (knownNat @32) BV.mkUnsignedBV unsigned
-  , testProperty "randomR signed well-formed and in bounds" $
+  , testPropertyString "randomR signed well-formed and in bounds" $
     testRandomR (knownNat @32) BV.mkSignedBV unsigned
   ]
 

